src/lib.rs

```diff
@@ -223,9 +223,29 @@ fn components(blurhash: &str) -> Result<(usize, usize), Error> {
     Ok((num_x, num_y))
 }
 
+#[cfg(feature = "image")]
+/// Calculates the blurhash for an [DynamicImage][image::DynamicImage] using the given x and y component counts.
+pub fn encode_image(components_x: u32, components_y: u32, image: &image::DynamicImage) -> Result<String, Error> {
+    use image::{GenericImageView, EncodableLayout};
+    encode(components_x, components_y, image.width(), image.height(), image.to_rgba8().as_bytes())
+}
+
+#[cfg(feature = "image")]
+/// Decodes the given blurhash to an image of the specified size.
+///
+/// The punch parameter can be used to de- or increase the contrast of the
+/// resulting image.
+pub fn decode_image(blurhash: &str, width: u32, height: u32, punch: f32) -> Result<image::DynamicImage, Error> {
```

Comment:
```suggestion
/// Decodes the given blurhash to an image of the specified size.
///
/// The punch parameter can be used to de- or increase the contrast of the
/// resulting image.
pub fn decode_image(blurhash: &str, width: u32, height: u32, punch: f32) -> Result<image::DynamicImage, Error> {
#[cfg(feature = "image")]
```

Additionally, I'm wondering whether methods for static image types would be useful too, especially since we already decode to `RgbaImage`.